ledger/ledger-common/src/main/java/net/corda/v5/ledger/common/transaction/TransactionSignatureService.java

```diff
@@ -57,12 +58,21 @@ List<List<DigitalSignatureAndMetadata>> signBatch(
      * The underlying verification service signals the verification failures with different exceptions.
      * {@link DigitalSignatureVerificationService}
      *
-     * @param transaction The original transaction.
+     * @param transaction           The original transaction.
      * @param signatureWithMetadata The signature to be verified.
+     * @param publicKey             The public key to verify against. It should match with signatureWithMetadata's keyId.
      * @throws RuntimeException if the signature could not be verified.
      */
     void verifySignature(
             @NotNull final TransactionWithMetadata transaction,
-            @NotNull final DigitalSignatureAndMetadata signatureWithMetadata
+            @NotNull final DigitalSignatureAndMetadata signatureWithMetadata,
+            @NotNull final PublicKey publicKey
+    );
+
+    // TODO Not too sure how good place is this for this function, but since Transactionsignature service is already available where this will be used
+    // in signed txs etc, probable good enough for now.
+    SecureHash getIdOfPublicKey(
+            @NotNull final PublicKey publicKey,
+            @NotNull final String digestAlgorithmName
```

Comment:
should be nullable in which case platform default digest (as given by DigestService, but we seem to still be missing the ability to get and digest algorithm names anywhere!)

Reply:
Speaking to @kyriathar he notes that the algorithm should match that of the id you are comparing to. So perhaps the method should be a matchesId(id: SecureHash) on PublicKey